fix: impl PartialEq<str> for CompactStr

DonIsaac opened the pull request:
I need this for another PR that I'm working on, but I think adding this just makes sense.

overlookmotel commented:
Nice. It has always been an annoyance that you can't compare `&compact_str == "foo"`.

But do you want to also add the reverse comparison `impl PartialEq<CompactStr> for str`? This PR at present makes `&compact_str == "foo"` valid, but not `"foo" == &compact_str`.

DonIsaac commented:
> `impl PartialEq<CompactStr> for str`?

Done

overlookmotel commented:
Thanks Don!

DonIsaac committed:
fix: impl PartialEq<str> for CompactStr (#4352)

I need this for another PR that I'm working on, but I think adding this just makes sense.
